core/plugins/org.polarsys.capella.core.data.interaction.validation/src/org/polarsys/capella/core/data/interaction/validation/scenario/InstanceRoleHasDifferentNameThanRepresentedInstance.java

```diff
@@ -0,0 +1,130 @@
+ * Contributors:
+ *    Thales - initial API and implementation
+ *******************************************************************************/
+package org.polarsys.capella.core.data.interaction.validation.scenario;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.emf.common.util.EList;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.impl.BasicEObjectImpl;
+import org.eclipse.emf.validation.EMFEventType;
+import org.eclipse.emf.validation.IValidationContext;
+import org.eclipse.emf.validation.model.ConstraintStatus;
+import org.eclipse.emf.validation.service.ConstraintRegistry;
+import org.eclipse.emf.validation.service.IConstraintDescriptor;
+import org.polarsys.capella.common.helpers.EObjectLabelProviderHelper;
+import org.polarsys.capella.core.data.information.AbstractInstance;
+import org.polarsys.capella.core.data.interaction.InstanceRole;
+import org.polarsys.capella.core.data.interaction.Scenario;
+import org.polarsys.capella.core.sirius.analysis.tool.StringUtil;
+import org.polarsys.capella.core.validation.rule.AbstractValidationRule;
+
+public class InstanceRoleHasDifferentNameThanRepresentedInstance extends AbstractValidationRule {
+  /**
+   * A container class that encloses all the elements required to efficiently perform the quick fix for the rename.</br>
+   * This container will be added to the validation context and used in the quick fix. </br>
+   * We are required to inherit from EObject since the validation context can contain only EObjects.
+   */
+  public static final class InstanceRoleRenamingData extends BasicEObjectImpl {
+    public final String newName;
+    public final InstanceRole instanceRole;
+
+    public InstanceRoleRenamingData(InstanceRole instanceRole, String newName) {
+      this.instanceRole = instanceRole;
+      this.newName = newName;
+    }
+  }
+
+  public static final String ID = "org.polarsys.capella.core.data.interaction.validation.DWF_DS_24"; // $NON-NLS-N$
+
+  @Override
+  public IStatus validate(IValidationContext validationContext) {
+    EObject target = validationContext.getTarget();
+    EMFEventType eventType = validationContext.getEventType();
+    List<IStatus> failureStatuses = new ArrayList<>();
+
+    if (eventType == EMFEventType.NULL && target instanceof Scenario) {
+      Scenario scenario = (Scenario) target;
+      Map<AbstractInstance, Collection<InstanceRole>> representedInstanceToInstanceRoleMap = //
+          extractRepresentedInstanceToInstanceRoleMap(scenario);
+
+      for (Map.Entry<AbstractInstance, Collection<InstanceRole>> entry : representedInstanceToInstanceRoleMap
+          .entrySet()) {
+        AbstractInstance representedInstance = entry.getKey();
+        String representedInstanceName = representedInstance.getName();
+
+        if (!StringUtil.isNullOrEmpty(representedInstanceName)) {
```

Comment:
if the user set to empty the represented instance, then the validation rule will not raise an error

Reply:
Another validation rule will then be raised
![image](https://user-images.githubusercontent.com/34767327/98348859-8dc98e80-2019-11eb-9736-79e75fe477fc.png)